18F/identity-idp
# LG-15321: relax vendor redirect in test mode

## 🎫 Ticket
[LG-15321](https://cm-jira.usa.gov/browse/LG-15321)

## 🛠 Summary of changes
Use a flag to disable redirect_to_correct_vendor filter.

## 📜 Testing Plan
When redirect_to_correct_vendor is enforced
1.  Set `doc_auth_redirect_to_correct_vendor_disabled` to false in application.yml
2. While on Socure document capture page, visit Mock/LN  document capture page  and verify Socure  document capture page is displayed
3. While on Socure hybrid document capture page, visit Mock/LN hybrid  document capture page and verify Socure hybrid document capture page is displayed
4. While on Mock/LN document capture page, visit Socure  document capture page  and verify Mock/LN  document capture page is displayed
5. While on Mock/LN hybrid document capture page, visit Socure hybrid document capture page  and verify Mock/LN  hybrid document capture page is displayed


When redirect_to_correct_vendor is relaxed
1.  Set `doc_auth_redirect_to_correct_vendor_disabled` to true in application.yml
2. While on Socure document capture page, visit Mock/LN  document capture page  and verify Mock/LN  document capture page is displayed
3. While on Socure hybrid document capture page, visit Mock/LN hybrid  document capture page and verify Mock/LN hybrid document capture page is displayed
4. While on Mock/LN document capture page, visit Socure  document capture page  and verify Socure  document capture page is displayed
5. While on Mock/LN hybrid document capture page, visit Socure hybrid document capture page  and verify Socure  hybrid document capture page is displayed
